tests/reduction/reduction_get_lambda.h

```diff
@@ -0,0 +1,269 @@
+ *         constructing reducer with span
+ *  @retval Lambda with chosen operator
+ */
+template <typename VariableT, typename RangeT, typename UseCombineFlagT,
+          typename FunctorT = void, typename AccessorT>
+auto get_lambda(AccessorT accessor) {
+  if constexpr (std::is_same<RangeT, sycl::range<1>>::value) {
+    return get_lambda_with_range<VariableT, FunctorT, UseCombineFlagT>(
+        accessor);
+  } else if constexpr (std::is_same<RangeT, sycl::nd_range<1>>::value) {
+    return get_lambda_with_nd_range<VariableT, FunctorT, UseCombineFlagT>(
+        accessor);
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer while
+ *         sycl::handler.parallel_for used sycl::range
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
+ *          calling .combine() function or operator +, *, ^= e.t.c.
+ *  @tparam AccessorT buffer accessor type
+ *  @param accessor Accessor to the buffer
+ *  @param number_elements Number elements in sycl::span, used only when
+ *         constructing reducer with span
+ *  @retval Lambda with chosen operator
+ */
+template <typename VariableT, typename FunctorT, typename UseCombineFlagT,
+          typename AccessorT>
+auto get_lambda_with_range_for_span(AccessorT accessor,
+                                    size_t number_elements) {
+  if constexpr (UseCombineFlagT::value) {
+    return [=](sycl::id<1> idx, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i].combine(accessor[idx]);
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT,
+                                    sycl::multiplies<VariableT>>::value) {
+    return [=](sycl::id<1> idx, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] *= accessor[idx];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT, sycl::plus<VariableT>>::value) {
+    return [=](sycl::id<1> idx, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] += accessor[idx];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT, sycl::bit_or<VariableT>>::value) {
+    return [=](sycl::id<1> idx, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] |= accessor[idx];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT,
+                                    sycl::bit_and<VariableT>>::value) {
+    return [=](sycl::id<1> idx, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] &= accessor[idx];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT,
+                                    sycl::bit_xor<VariableT>>::value) {
+    return [=](sycl::id<1> idx, auto &reducer) {
+      for (size_t i = 0; i < number_elements; i++) {
+        reducer[i] ^= accessor[idx];
```

Comment:
Factorize out all the loops. Use just 1 loop with all `if constexpr` in it instead.
Probably all these `if constexpr` patterns should be also factorized into a single generic `operation_dispatch` to be used as a functional *lord of the rings* for these copy-pasted dispatching patterns.

Reply:
If after implementing this commit: 823e2fbe3bded9166662dc948031fd61294f4e36 this question still open, please, let me know